### PR TITLE
SNOW-1011778: Adding 'spcs service upgrade' command

### PR DESCRIPTION
**Pull request by sfc-gh-davwang**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Added 'spcs service upgrade' command which allows you to update the spec of a running service.

**Usage**:
image

### CODE REVIEW

File: src/snowflake/cli/plugins/spcs/services/manager.py [hunk 1]
```diff
@@ -83,3 +83,13 @@ def logs(
...
+
+    def upgrade_spec(self, service_name: str, spec_path: Path):
+        spec = self._read_yaml(spec_path)
+        query = f"""
+        alter service {service_name}
+        from specification $$
+        {spec}
+        $$
+        """.splitlines()
+        return self._execute_schema_query(strip_empty_lines(query))
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/spcs/services/manager.py:95]: Why we need splitlines  and strip_empty lines? AFAIK we don't use it in other places

Comment (sfc-gh-davwang) [reply src/snowflake/cli/plugins/spcs/services/manager.py:95]: The intention was to remove white space and blank lines from a multi line query. Previously the query had been constructed in parts so I used strip_empty_lines, however you are right that now a simple .strip() would be enough to remove trailing/leading whitespace. I've made the change.

**Review by sfc-gh-yiwang — APPROVED** [review]
LGTM

**Review by sfc-gh-cgorrie — APPROVED** [review]
`project/test_util.py` changes LGTM

File: tests/spcs/test_services.py [hunk 2]
```diff
@@ -322,3 +302,47 @@ def test_resume_cli(mock_resume, mock_cursor, runner):
...
+    mock_upgrade_spec.return_value = cursor
+    service_name = "test_service"
+    tmp_dir = Path(other_directory)
+    spec_path = tmp_dir / "spec.yml"
+    spec_path.write_text(SPEC_CONTENT)
+
+    result = runner.invoke(
+        ["spcs", "service", "upgrade", service_name, "--spec-path", spec_path]
+    )
+
```
Comment (sfc-gh-davwang) [inline tests/spcs/test_services.py:346]: add called_once_with assertion

File: tests_integration/testing_utils/snowpark_services_utils.py [hunk 3]
```diff
@@ -143,6 +136,40 @@ def _wait_until_service_reaches_state(
         ).strip()
         pytest.fail(error_message)
 
+    def upgrade_service(self, service_name: str):
+        describe_result = self._execute_describe(service_name)
```
Comment (sfc-gh-davwang) [inline tests_integration/testing_utils/snowpark_services_utils.py:140]: maybe upgrade_service_should_change_spec

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**